app_test_suite/steps/base.py

```diff
@@ -0,0 +1,173 @@
+            action="store_true",
+            help="Skip automated app deployment for the test run to the test cluster (using an App CR).",
+        )
+        self._config_parser_group.add_argument(
+            self.KEY_CONFIG_OPTION_DEPLOY_NAMESPACE,
+            required=False,
+            default="default",
+            help="The namespace your app under test should be deployed to for running tests.",
+        )
+        self._config_parser_group.add_argument(
+            self.KEY_CONFIG_OPTION_DEPLOY_CONFIG_FILE,
+            required=False,
+            help="Path for a configuration file (values file) for your app when it's deployed for testing.",
+        )
+        self._cluster_manager.initialize_config(self._config_parser_group)
+
+    def pre_run(self, config: argparse.Namespace) -> None:
+        super().pre_run(config)
+        if self._all_pre_runs_skipped:
+            return
+
+        if not config.chart_file or not os.path.isfile(config.chart_file):
+            raise ConfigError("chart-file", f"The file '{config.chart_file}' can't be found.")
+
+        self._cluster_manager.pre_run(config)
+        app_config_file = get_config_value_by_cmd_line_option(config, self.KEY_CONFIG_OPTION_DEPLOY_CONFIG_FILE)
+        if app_config_file:
+            if not os.path.isfile(app_config_file):
+                raise ATSTestError(
+                    f"Application test run was configured to use '{app_config_file}' as app"
+                    f" config file, but it doesn't exist."
+                )
+            try:
+                with open(app_config_file, "r") as file:
+                    yaml.safe_load(file)
+            except Exception:
+                raise ATSTestError(
+                    f"Application config file '{app_config_file}' found, but can't be loaded"
+                    f"as a correct YAML document."
+                )
+
+    def cleanup(
+        self,
+        config: argparse.Namespace,
+        context: Context,
+        has_build_failed: bool,
+    ) -> None:
+        self._cluster_manager.cleanup()
+
+
+@dataclass
+class TestExecInfo:
+    """
+    TestExecInfo provides all the information that is passed from a test scenario to test executor.
+    """
+
+    chart_path: str
+    """Path to the chart file under test."""
+    chart_ver: str
+    """Chart version detected from the chart."""
+    app_config_file_path: Optional[str]
+    """Path to an optional Helm values file used to configure the chart under test."""
+    cluster_type: str
+    """A string representing a cluster type that the test scenario is running on."""
+    cluster_version: str
+    """Kubernetes cluster version of a cluster the test scenario is running on."""
+    kube_config_path: str
+    """Path to kube.config to connect to the cluster."""
+    test_type: str
+    """Type of test to execute by the test executor."""
+    test_dir: str
+    """Path to a directory where test code is available."""
+
+
+class TestExecutor(ABC):
```

Comment:
ABC is short for Abstract Build Class?

Name is cool but I find the naming a bit cryptic.